### PR TITLE
Fix the plugin setup

### PR DESCRIPTION
**Pull request by ndaelman-hu**
Follow the example in https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/merge_requests/1655/

Before merging, make sure to add the full extraction.

### CODE REVIEW

File: src/nomad_parser_vasp/parsers/xml_parser.py [hunk 1]
```diff
@@ -128,19 +60,7 @@ def parse(
...
-                    model_system=[ModelSystem(cell=AtomicCell)],
-                    model_method=[DFT, KMesh],
-                    output=[ElectronicEigenvalues],
-                ),
-            ],
-        )
-        xml_parser = XMLParser(filepath='vasprun.xml')  # TODO apply match
-        xml_parser.convert(archive_parser)
-        archive.data = archive_parser.data_object
+        logger.info('VasprunXMLParser.parse', parameter=configuration.parameter)
```
Comment (ladinesa) [inline src/nomad_parser_vasp/parsers/xml_parser.py:29]: if this is a matching parsee you would meed to.add parsee specs

**Comment by ladinesa** [comment]
I think this shouldalready work.

File: src/nomad_parser_vasp/parsers/xml_parser.py [hunk 2]
```diff
@@ -1,123 +1,55 @@
...
-    xml=MappingAnnotationModel(
-        path=ElectronicEigenvalues.spin_channel.m_annotations.xml
-        + '.set[?"@comment"="kpoint 1"]'
-    )  # TODO not going to work: add conversion to reference
+Simulation.m_def.m_annotations['xml'] = MappingAnnotationModel(path='modeling')
+
+Simulation.program.m_annotations['xml'] = MappingAnnotationModel(path='.generator')
+
+
+Program.name.m_annotations['xml'] = MappingAnnotationModel(
```
Comment (ladinesa) [inline src/nomad_parser_vasp/parsers/xml_parser.py:40]: you are mussing xml here

**Comment by ndaelman-hu** [comment]
> I think this shouldalready work.

Me too. `Simulation` has the proper `m_annotations` and `XMLParser` reads in all the tags...
But after `convert`, `Simulation` remains empty.

File: src/nomad_parser_vasp/parsers/xml_parser.py [hunk 3]
```diff
@@ -1,123 +1,55 @@
...
+    path='.i[?"@name"="program"]'
```
Comment (ladinesa) [inline src/nomad_parser_vasp/parsers/xml_parser.py]: pls try to use simgle quotes inside somerimes jnespath is crazy and it should be ==

File: src/nomad_parser_vasp/parsers/xml_parser.py (hunk 1 above)
Comment (ndaelman-hu) [reply src/nomad_parser_vasp/parsers/xml_parser.py:29]: You mean like the filename to match? I already added it to `XMLParser`

File: src/nomad_parser_vasp/parsers/xml_parser.py [hunk 4]
```diff
@@ -1,123 +1,55 @@
...
 )
-ElectronicEigenvalues.occupation.m_annotations = dict(
-    xml=MappingAnnotationModel(
-        path=ElectronicEigenvalues.reciprocal_cell.m_annotations.xml + '.r[0]'
-    )
+
+
+Program.version.m_annotations['xml'] = MappingAnnotationModel(
+    path='.i[?"@name"="version"]'
```
Comment (ladinesa) [inline src/nomad_parser_vasp/parsers/xml_parser.py]: also here

File: src/nomad_parser_vasp/parsers/xml_parser.py [hunk 5]
```diff
@@ -1,123 +1,55 @@
...
 )
-ElectronicEigenvalues.value.m_annotations = dict(
-    xml=MappingAnnotationModel(
-        path=ElectronicEigenvalues.reciprocal_cell.m_annotations.xml + '.r[1]'
-    )
+
+# ? compilation mode
+Program.compilation_host.m_annotations['xml'] = MappingAnnotationModel(
+    path='.i[?"@name"="platform"]'
```
Comment (ladinesa) [inline src/nomad_parser_vasp/parsers/xml_parser.py]: alao here

**Comment by ndaelman-hu** [comment]
Mostly fixed, but have to close this branch for the upcoming tutorial. Will revisit in the future.

**Pull request merged**